+ source ../common/common_lib.sh
++ source /root/mugen-riscv/libs/locallibs/common_lib.sh
+++ python3 --version
+++ '[' 0 -eq 0 ']'
+++ source /root/mugen-riscv/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Sun Dec  4 03:13:03 2022 - INFO  - Start environmental preparation.
+ DNF_INSTALL pesign
+ pkgs=pesign
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py install --pkgs pesign --node 1 --tempfile ''
+ tmpfile2=/tmp/tmp1y90vymv
+ '[' -z '' ']'
+ tmpfile=/tmp/tmp1y90vymv
+ mkdir /etc/pki/pesign
mkdir: cannot create directory ‘/etc/pki/pesign’: File exists
+ expect
spawn certutil -d /etc/pki/pesign/ -N
Enter Password or Pin for "NSS Certificate DB":+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Sun Dec  4 03:14:37 2022 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start testing...'
+ message='Start testing...'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Start testing...'
Sun Dec  4 03:14:38 2022 - INFO  - Start testing...
+ test_execution pesign.service
+ service=pesign.service
++ date '+%Y-%m-%d %T'
+ log_time='2022-12-04 03:14:38'
+ test_restart pesign.service
+ service=pesign.service
+ systemctl restart pesign.service
Job for pesign.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status pesign.service" and "journalctl -xeu pesign.service" for details.
+ CHECK_RESULT 1 0 0 'pesign.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='pesign.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'pesign.service restart failed'
+ LOG_ERROR 'pesign.service restart failed'
+ message='pesign.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'pesign.service restart failed'
Sun Dec  4 03:14:42 2022 - ERROR - pesign.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 34'
+ message='../common/common_lib.sh line 34'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 34'
Sun Dec  4 03:14:43 2022 - ERROR - ../common/common_lib.sh line 34
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status pesign.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'pesign.service restart failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='pesign.service restart failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'pesign.service restart failed'
+ LOG_ERROR 'pesign.service restart failed'
+ message='pesign.service restart failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'pesign.service restart failed'
Sun Dec  4 03:14:50 2022 - ERROR - pesign.service restart failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 37'
+ message='../common/common_lib.sh line 37'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 37'
Sun Dec  4 03:14:51 2022 - ERROR - ../common/common_lib.sh line 37
+ return 0
+ systemctl stop pesign.service
+ CHECK_RESULT 0 0 0 'pesign.service stop failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='pesign.service stop failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status pesign.service
+ grep 'Active: inactive'
+ CHECK_RESULT 1 0 0 'pesign.service stop failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='pesign.service stop failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'pesign.service stop failed'
+ LOG_ERROR 'pesign.service stop failed'
+ message='pesign.service stop failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'pesign.service stop failed'
Sun Dec  4 03:14:58 2022 - ERROR - pesign.service stop failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 42'
+ message='../common/common_lib.sh line 42'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 42'
Sun Dec  4 03:14:58 2022 - ERROR - ../common/common_lib.sh line 42
+ return 0
+ systemctl start pesign.service
Job for pesign.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status pesign.service" and "journalctl -xeu pesign.service" for details.
+ CHECK_RESULT 1 0 0 'pesign.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='pesign.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'pesign.service start failed'
+ LOG_ERROR 'pesign.service start failed'
+ message='pesign.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'pesign.service start failed'
Sun Dec  4 03:15:03 2022 - ERROR - pesign.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 44'
+ message='../common/common_lib.sh line 44'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 44'
Sun Dec  4 03:15:03 2022 - ERROR - ../common/common_lib.sh line 44
+ return 0
+ SLEEP_WAIT 5
+ wait_time=5
+ cmd=
+ mode=1
+ python3 /root/mugen-riscv/libs/locallibs/sleep_wait.py --time 5 --cmd '' --mode 1
+ systemctl status pesign.service
+ grep 'Active: active'
+ CHECK_RESULT 1 0 0 'pesign.service start failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='pesign.service start failed'
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'pesign.service start failed'
+ LOG_ERROR 'pesign.service start failed'
+ message='pesign.service start failed'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'pesign.service start failed'
Sun Dec  4 03:15:10 2022 - ERROR - pesign.service start failed
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 47'
+ message='../common/common_lib.sh line 47'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 47'
Sun Dec  4 03:15:11 2022 - ERROR - ../common/common_lib.sh line 47
+ return 0
+ test_enabled pesign.service
+ service=pesign.service
++ systemctl is-enabled pesign.service
+ state=disabled
+ '[' disabled == enabled ']'
+ '[' disabled == disabled ']'
++ systemctl enable pesign.service
++ grep 'Created symlink'
++ awk '{print $3}'
+ symlink_file=/etc/systemd/system/multi-user.target.wants/pesign.service
+ find /etc/systemd/system/multi-user.target.wants/pesign.service
/etc/systemd/system/multi-user.target.wants/pesign.service
+ CHECK_RESULT 0 0 0 'pesign.service enable failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='pesign.service enable failed'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl disable pesign.service
Removed /etc/systemd/system/multi-user.target.wants/pesign.service.
+ find /etc/systemd/system/multi-user.target.wants/pesign.service
find: ‘/etc/systemd/system/multi-user.target.wants/pesign.service’: No such file or directory
+ CHECK_RESULT 1 0 1 'pesign.service disable failed'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='pesign.service disable failed'
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ journalctl --since '2022-12-04 03:14:38' -u pesign.service
+ grep -i 'fail\|error'
+ grep -v -i 'DEBUG\|INFO\|WARNING'
Dec 04 03:14:41 anolis-riscv systemd[1]: pesign.service: Failed with result 'core-dump'.
Dec 04 03:14:41 anolis-riscv systemd[1]: Failed to start pesign.service - Pesign signing daemon.
Dec 04 03:15:02 anolis-riscv systemd[1]: pesign.service: Failed with result 'core-dump'.
Dec 04 03:15:02 anolis-riscv systemd[1]: Failed to start pesign.service - Pesign signing daemon.
+ CHECK_RESULT 0 0 1 'There is an error message for the log of pesign.service'
+ actual_result=0
+ expect_result=0
+ mode=1
+ error_log='There is an error message for the log of pesign.service'
+ '[' -z 0 ']'
+ '[' 1 -eq 0 ']'
+ test 0x == 0x
+ test -n 'There is an error message for the log of pesign.service'
+ LOG_ERROR 'There is an error message for the log of pesign.service'
+ message='There is an error message for the log of pesign.service'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'There is an error message for the log of pesign.service'
Sun Dec  4 03:15:19 2022 - ERROR - There is an error message for the log of pesign.service
+ (( exec_result++ ))
+ LOG_ERROR '../common/common_lib.sh line 28'
+ message='../common/common_lib.sh line 28'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message '../common/common_lib.sh line 28'
Sun Dec  4 03:15:20 2022 - ERROR - ../common/common_lib.sh line 28
+ return 0
+ test_reload pesign.service
+ service=pesign.service
+ systemctl start pesign.service
Job for pesign.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status pesign.service" and "journalctl -xeu pesign.service" for details.
+ systemctl reload pesign.service
+ grep 'Job type reload is not applicable'
Failed to reload pesign.service: Job type reload is not applicable for unit pesign.service.
+ CHECK_RESULT 0 0 0 'Job type reload is not applicable for unit pesign.service'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Job type reload is not applicable for unit pesign.service'
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ systemctl status pesign.service
+ grep 'Active: active'
+ systemctl status pesign.service
+ grep 'inactive (dead)'
+ return 1
+ LOG_INFO 'Finish test!'
+ message='Finish test!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish test!'
Sun Dec  4 03:15:25 2022 - INFO  - Finish test!
+ CASE_RESULT 0
+ case_re=0
+ test -z 6
+ test 6 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Sun Dec  4 03:15:26 2022 - ERROR - failed to execute the case.
+ exit 6
+ post_test
+ LOG_INFO 'start environment cleanup.'
+ message='start environment cleanup.'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'start environment cleanup.'
Sun Dec  4 03:15:27 2022 - INFO  - start environment cleanup.
+ systemctl stop pesign.service
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmp1y90vymv ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen-riscv/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmp1y90vymv
+ '[' 0 -ne 0 ']'
+ rm -rf /etc/pki/pesign log
+ LOG_INFO 'Finish environment cleanup!'
+ message='Finish environment cleanup!'
+ python3 /root/mugen-riscv/libs/locallibs/mugen_log.py --level info --message 'Finish environment cleanup!'
Sun Dec  4 03:15:42 2022 - INFO  - Finish environment cleanup!
